src/install/npm.zig

```diff
@@ -823,16 +823,26 @@ pub const PackageManifest = struct {
 
         {
             const releases = this.pkg.releases.keys.get(this.versions);
+
+            var best_match_version: ?Semver.Version = null;
+            var best_match_package_index: usize = 0;
             var i = releases.len;
             // For now, this is the dumb way
             while (i > 0) : (i -= 1) {
                 const version = releases[i - 1];
-                const packages = this.pkg.releases.values.get(this.package_versions);
 
-                if (group.satisfies(version)) {
-                    return .{ .version = version, .package = &packages[i - 1] };
+                // If we find one that matches, save it, but keep looping because we might find a newer match.
+                // The versions from the registry are not in any particular order.
+                if (group.satisfies(version) and (best_match_version == null or Semver.Version.order(version, best_match_version.?, "", "") == .gt)) {
```

Comment:
Do we need to account for pre/post tags? Usually the way we do this is to get the string buffer from `lockfile.buffers.string_bytes.items` and pass that to the parent function
 
```suggestion
                if (group.satisfies(version) and (best_match_version == null or Semver.Version.order(version, best_match_version.?, string_bytes, string_bytes) == .gt)) {
```